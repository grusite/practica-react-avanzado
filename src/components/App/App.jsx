import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'

import Home from '../Home/Home'
import Register from '../Register/Register'
import AdvertDetail from '../AdvertDetail/AdvertDetail'
import CreateUpdateAdvert from '../CreateUpdateAdvert/CreateUpdateAdvert'
import Error404 from '../Error404/Error404'

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props)
    this.state = { error: null }
  }

  componentDidCatch(error, errorInfo) {
    this.setState({ error })
  }

  render() {
    if (this.state.error) {
      return (
        <div className="snap">
          <div className="snap-message">
            <p>We're sorry - something's gone wrong.</p>
            <p>
              Our team has been notified, but click <button>here</button> to fill out a report.
            </p>
          </div>
        </div>
      )
    } else {
      return this.props.children
    }
  }
}

class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  // <Route component={Error404} />

  render() {
    return (
      <ErrorBoundary>
        <Router>
          <Switch>
            <Route path="/register" component={Register} />
            <Route exact path="/advert" component={Home} />
            <Route path="/advert/:id" component={AdvertDetail} />
            <Route path="/create" component={CreateUpdateAdvert} />
            <Route path="/update" component={CreateUpdateAdvert} />
            <Route exact path="/" component={Register} />
            <Route component={Error404} />
          </Switch>
        </Router>
      </ErrorBoundary>
    )
  }
}

export default App
