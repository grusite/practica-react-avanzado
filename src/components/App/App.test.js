import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
  const render = () => shallow(<App />);

  it("should render a App component", () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });
});
