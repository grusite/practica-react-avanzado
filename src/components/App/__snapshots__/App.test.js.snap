// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a App component 1`] = `
<ErrorBoundary>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Connect(Register))",
          "type": [Function],
        }
      }
      path="/register"
    />
    <Connect(PrivateRoute)
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(AdvertDetail)",
          "type": [Function],
        }
      }
      path="/advert/:id"
    />
    <Connect(PrivateRoute)
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Connect(createUpdateAdvert))",
          "type": [Function],
        }
      }
      path="/create"
    />
    <Connect(PrivateRoute)
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Connect(createUpdateAdvert))",
          "type": [Function],
        }
      }
      path="/update/:id"
    />
    <Connect(PrivateRoute)
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Connect(Home))",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Connect(PrivateRoute)
      component={[Function]}
    />
  </Switch>
</ErrorBoundary>
`;
